src/arvgvfakecamera.c

```diff
@@ -483,20 +500,46 @@ arv_gv_fake_camera_start (ArvGvFakeCamera *gv_fake_camera)
 			}
 			g_clear_object (&inet_socket_address);
 			g_clear_object (&inet_address);
-
-			g_free (gvcp_address_string);
+                        
+                        /* subnet broadcast discovery */
+			socket_address = g_socket_address_new_from_native (ifap_iter->ifa_broadaddr,
+									   sizeof (struct sockaddr));
+                        inet_address = g_inet_socket_address_get_address (G_INET_SOCKET_ADDRESS (socket_address));
+			discovery_address_string = g_inet_address_to_string (inet_address);
+			arv_debug_device ("[GvFakeCamera::start] Discovery address = %s", discovery_address_string);
 			g_free (discovery_address_string);
+			inet_socket_address = g_inet_socket_address_new (inet_address, ARV_GVCP_PORT);
+			if (g_strcmp0 (gvcp_address_string, discovery_address_string) == 0)
```

Comment:
Use of discovery_address_string after free

Reply:
Double oops!